packages/core/crmManager/crmManager.ts

```diff
@@ -0,0 +1,65 @@
+import getCrm from "@calcom/app-store/_utils/getCrm";
+import logger from "@calcom/lib/logger";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { CredentialPayload } from "@calcom/types/Credential";
+import type { CRM, ContactCreateInput } from "@calcom/types/CrmService";
+
+const log = logger.getSubLogger({ prefix: ["CrmManager"] });
+export default class CrmManager {
+  crmService: CRM | null | undefined = null;
+  credential: CredentialPayload;
+  constructor(credential: CredentialPayload) {
+    this.credential = credential;
+  }
+
+  private async getCrmService(credential: CredentialPayload) {
```

Comment:
Every public method will start by calling this. If it's the first time a method is called, get the appropriate `CrmService`

Reply:
Would it make sense to just call this in the CTOR of CRM manager to prevent us needing to call 

await this.getCrmService(this.credential);

In every method and future method.  I think adding this.getCrmService() into the ctor would work if we change the implementation to use `this.credential` instead of as an arg (as they are same)

See suggestion for an example

Reply:
I was thinking about this as well but after some reading I learned that constructors should only run synchronous code. I was considering the factory pattern but I think it would have been overkill for this and I want to keep this class similar to what we have now. This is like a pseudo-decorator pattern.

Reply:
I think we can change the approach to something like this atleast. It keeps the code almost same but still making it more intuitive.

![image](https://github.com/calcom/cal.com/assets/1780212/58223b1c-a78f-4919-9e0e-5cc16332d052)